### PR TITLE
Feature/enable 3rd parties

### PR DESCRIPTION
**Pull request by ProbablePrime**
This just wraps requires that we do that might not exist should a user not have access to backend to return {}.

Then in the pug we just guard against blank objects with some conditionals.

3rd parties wishing to contribute can now just do `gulp recompile` followed by `gulp html-quick/watch` to build the tutorial and reference sections.

### CODE REVIEW

File: src/rest/template.pug [hunk 1]
```diff
@@ -24,12 +24,13 @@ include ./resource.pug
                     th Request Count
                     th Time Interval (in Seconds)
             tbody
-                - orderered = orderObject(beConfig.ratelimit.buckets)
-                for bucket, name in orderered
-                    tr
-                        td= name
-                        td= bucket.max
-                        td= bucket.interval/1000
+                if beConfig.ratelimt && beConfig.ratelimt.buckets
```
Comment (UnwrittenFun) [inline src/rest/template.pug]: ratelimit

Comment (ProbablePrime) [inline src/rest/template.pug]: :scream:

**Comment by UnwrittenFun** [comment]
lgtm

File: build/misc.js [hunk 2]
```diff
@@ -15,7 +15,19 @@ const orderObject = require('./util').orderObject;
...
 function readJSONFile (filePath) {
-    return JSON.parse(fs.readFileSync(filePath));
+    try {
+        return JSON.parse(fs.readFileSync(filePath));
+    } catch (e) {
+        return {};
+    }
+}
+
+function wrappedRequire (filepath) {
```
Comment (SimonSchick) [inline build/misc.js]: docs

**Comment by ProbablePrime** [comment]
Docs are done @SimonSchick

File: src/rest/template.pug [hunk 3]
```diff
@@ -45,41 +46,44 @@ include ./resource.pug
...
-                .panel-body
-                    if resource.description
-                        .top-resource-description!= marked(resource.description)
-                    .panel-group
-                        +resourceModals(resource)
-                        table(style='width:100%')
-                            tbody
-                                +resourceRows(resource)
+        if rest.resources
+            for resource in rest.resources
```
Comment (SimonSchick) [inline src/rest/template.pug]: I'm not sure if I like this, I feel like it would be cleaner to define these as empty arrays in the fallback object.

**Comment by SimonSchick** [comment]
One last thing, lgtm otherwise.

**Comment by ProbablePrime** [comment]
Except that they are accessed as an object throughout the pug?

Null would probably be better than in the conditionals within the pug we can just wrap them in a top level conditional instead of doing nested guards down the object tree.

**Comment by SimonSchick** [comment]
sounds better

**Comment by ProbablePrime** [comment]
@SimonSchick sorted

**Comment by SimonSchick** [comment]
works for me

**Pull request merged**